### PR TITLE
Improvements to DynamoDBLogStore.java feature  #1

### PR DESCRIPTION
**Pull request by scottsand-db**
- javadocs
- fix error where `get` was using the file's json absolute path, but `put` was using the file name. using the fileName is correct.
- style improvements

### CODE REVIEW

File: storage-dynamodb/src/test/java/io/delta/storage/MemoryLogStore.java [hunk 1]
```diff
@@ -1,57 +1,89 @@
...
-    protected ExternalCommitEntry getLatestExternalEntry(Path tablePath) throws IOException {
+    protected Optional<ExternalCommitEntry> getLatestExternalEntry(Path tablePath) {
         return hashMap
             .values()
             .stream()
             .filter(item -> item.tablePath.equals(tablePath))
-            .sorted((a, b) -> b.absoluteJsonPath().compareTo(a.absoluteJsonPath()))
-            .findFirst().orElse(null);
+            .sorted((a, b) -> b.absoluteFilePath().compareTo(a.absoluteFilePath()))
+            .findFirst();
```
Comment (scottsand-db) [inline storage-dynamodb/src/test/java/io/delta/storage/MemoryLogStore.java:67]: TODO check that this returns the latest element

File: storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java [hunk 2]
```diff
@@ -50,97 +53,125 @@
...
             writeCopyTempFile(fs, entry.absoluteTempPath(), resolvedPath);
+
+            // Step 4: ACKNOWLEDGE the commit
             writePutCompleteDbEntry(entry);
-        } catch(Throwable e) {
-            LOG.info(String.format(
-                "%s: ignoring recoverable error: %s", e.getClass().getSimpleName(), e));
+        } catch (Throwable e) {
+            LOG.info(
+                "{}: ignoring recoverable error: {}", e.getClass().getSimpleName(), e
```
Comment (mrk-its) [inline storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java:168]: for some reason in logs I see:
```
22/03/30 18:52:10 INFO BaseExternalLogStore: IOException: ignoring recoverable error: {}
java.io.IOException: injected failure: write_copy_temp_file
	at io.delta.storage.FailingDynamoDBLogStore.injectError(FailingDynamoDBLogStore.java:77)
	at io.delta.storage.FailingDynamoDBLogStore.writeCopyTempFile(FailingDynamoDBLogStore.java:52)

```

Comment (mrk-its) [reply storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java:168]: notice `{}` placeholder in above message

Comment (scottsand-db) [reply storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java:168]: > Also whole stacktrace is present

Is that bad?

Comment (mrk-its) [reply storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java:168]: Placeholder `{}` in logs is probably unexpected. Stacktrace is probably ok, I'm only asking if it is intentional

Comment (scottsand-db) [reply storage-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java:168]: good catch! thanks

**Pull request merged**